Stratis.Bitcoin/BlockPulling/BlockPuller.cs

```diff
@@ -26,14 +64,41 @@ public abstract class BlockPuller : IBlockPuller
         /// <summary>Minimal quality score of a peer node based on the node's past experience with the peer node.</summary>
         public const int MinQualityScore = 1;
 
-        /// <summary>List of relations to peer nodes mapped by block header hashes that the peers are requested to provide.</summary>
-        private readonly ConcurrentDictionary<uint256, BlockPullerBehavior> map;
+        /// <summary>Instance logger.</summary>
+        protected readonly ILogger logger;
+
+        /// <summary>Lock protecting access to <see cref="map"/>, <see cref="pendingInventoryVectors"/>, <see cref="downloadedBlocks"/>, and <see cref="pendingDownloads"/></summary>
+        private readonly object lockObject = new object();
+
+        /// <summary>
+        /// List of relations to peer nodes mapped by block header hashes that the peers are requested to provide.
+        /// </summary>
+        /// <remarks>All access to this object has to be protected by <see cref="lockObject"/>.</remarks>
+        private readonly Dictionary<uint256, BlockPullerBehavior> map;
```

Comment:
yes, i used map as it was in the original code, but you are right it should be renamed, and i'll change the comment as well

i would avoid "mappings" suffix and i tend to call this "download tasks" in the comments, so i would go for "blockDownloadTasks" or "assignedBlockTasks", maybe the later is better (for me), 

the comment could be 

"Hashes of blocks to be downloaded mapped by the peers that the download tasks are assigned to."

what do you think?